aws/resource_aws_ec2_client_vpn_endpoint_test.go

```diff
@@ -111,6 +111,9 @@ func TestAccAwsEc2ClientVpn_serial(t *testing.T) {
 		for name, tc := range m {
 			tc := tc
 			t.Run(fmt.Sprintf("%s_%s", group, name), func(t *testing.T) {
+				t.Cleanup(func() {
+					testAccEc2ClientVpnEndpointSemaphore.Notify()
+				})
```

Comment:
This is currently deadlocking during normal `go test` (non acceptance testing) since the underlying function is waiting to close a channel that never gets opened. As a first line of defense we can guard this for acceptance testing:

```suggestion
				t.Cleanup(func() {
					if os.Getenv(resource.TestEnvVar) != "" {
						testAccEc2ClientVpnEndpointSemaphore.Notify()
					}
				})
```

But I'm pretty sure this can still fail if `testAccPreCheck(t)` fails in the current test cases. We can try switching the `PreCheck` handlers in the test cases to workaround that particular issue as well (annoying as that is logic-wise):

```go
PreCheck:     func() { testAccPreCheckClientVPNSyncronize(t); testAccPreCheck(t) },
```